cli/tests/fixtures/extensions/async-borrows/main.ts

```diff
@@ -0,0 +1,13 @@
+import * as PhylumApi from './phylum-api.ts'
+
+await Promise.all([
+  PhylumApi.get_user_info(),
+  PhylumApi.get_user_info(),
+  PhylumApi.get_user_info(),
+  PhylumApi.get_user_info(),
+  PhylumApi.get_user_info(),
+  PhylumApi.get_user_info(),
+  PhylumApi.get_user_info(),
+  PhylumApi.get_user_info(),
+  PhylumApi.get_user_info(),
```

Comment:
Maybe a tiny bit excessive? Also this sends actual web requests, I don't think we want that?

Reply:
The point of the test was to create a complex concurrent situation where there is some contention/risk of deadlocking.

This is still far from perfect as we can't _guarantee_ deterministic behavior, i.e. reliably reproducing a situation where a concurrency issue may occur in the first place.

I think actual calls are fine as long as we're in the context of an integration test -- the actual network access is part of the behavior under test. If we don't want that, we can mark this test as `#[ignore]` and make it so it only runs locally, but I do think having coverage on this is important.

Reply:
> The current one doesn't, but that doesn't mean that it is 100% correct and there is no corner case where concurrency issues happen. A shotgun approach is not elegant, I agree; what would be a good test here to have in its place?

There is no "100% guarantee" unless you want to start formally verifying our code. Tests should be purpose-driven and if you have an edge-case beyond our original problem that you think is uncovered, we cover it explicitly.